test/105_frag_2_test.sh

```diff
@@ -0,0 +1,30 @@
+#! /bin/bash
+
+. ./config.sh
+
+# 700 = Linux's minimum PMTU (552) + weave framing overheads
+# (currently about 60, but we leave some margin). That way the PMTUs
+# seen by containers is still >= 552.
+drop_large_udp_packets="FORWARD -p udp --destination-port 6783 -m length '!' --length 0:700 -j DROP"
+
+start_suite "PMTU discovery and packet fragmentation"
+
+run_on $HOST2 "sudo iptables -I $drop_large_udp_packets"
+
+weave_on $HOST1 launch
+weave_on $HOST2 launch $HOST1
+
+start_container $HOST1 --name=c1
+start_container $HOST2 --name=c2
+
+sleep 30 # give weave time to discover the PMTU
+
+# Check large packets get through. The first attempt typically fails,
+# since the sending container hasn't discovered the PMTU yet. The 2nd
+# attempt should succeed.
+exec_on $HOST2 c2 $PING -s 10000 c1 2>&1 1>/dev/null || true
+assert_raises "exec_on $HOST2 c2 $PING -s 10000 c1"
+
+run_on $HOST2 "sudo iptables -D $drop_large_udp_packets"
```

Comment:
> If something were to fail above ... the iptables entry will end up affecting other hosts? Perhaps we should clear any iptables runs in start_suite?

Can't really do the latter, since we have no idea what rules are meant to be there vs were inserted by us.

I've changed the tests so that now nothing can fail (recall that a failing assert does not cause the script to terminate). It's not bullet proof, e.g. the rule will still be left behind if you run with `STOP=1` and the test fails, or when something kills the test, but the alternatives seem too much trouble than it is worth.